internal/images/images.go

```diff
@@ -386,6 +390,78 @@ func decodeHEIFConfig(rat io.ReaderAt) (image.Config, error) {
 	}, nil
 }
 
+// parseLeadingNumber strips any unit suffix and converts the leading
+// number to float64.  Returns ok=false if it can’t.
+func parseLeadingNumber(s string) (v float64, ok bool) {
+	s = strings.TrimSpace(s)
+	i := 0
+
+	for _, r := range s {
+		if unicode.IsDigit(r) || r == '+' || r == '-' || r == '.' {
+			i++
+		} else {
+			break
+		}
+	}
+	if i == 0 {
+		return 0, false
+	}
+	n, err := strconv.ParseFloat(s[:i], 64)
+	return n, err == nil
+}
+
+// dimsFromSVG reads tokens until it sees the first <svg>
+// start-element, then extracts width/height/viewBox.
+func dimsFromSVG(r io.Reader) (w, h float64, err error) {
+	dec := xml.NewDecoder(r)
+	dec.Strict = false
+
+	for {
+		tok, errTok := dec.Token()
+		if errTok != nil {
+			if errTok == io.EOF {
+				return 0, 0, errors.New("no <svg> element found")
+			}
+			return 0, 0, errTok
+		}
+
+		se, ok := tok.(xml.StartElement)
+		if !ok || se.Name.Local != "svg" {
+			continue // not the root <svg> yet
+		}
+
+		var widthAttr, heightAttr, viewBoxAttr string
+		for _, a := range se.Attr {
+			switch a.Name.Local {
+			case "width":
+				widthAttr = a.Value
+			case "height":
+				heightAttr = a.Value
+			case "viewBox":
+				viewBoxAttr = a.Value
+			}
+		}
+
+		// Try width & height first
+		if w, ok = parseLeadingNumber(widthAttr); ok {
+			if h, ok = parseLeadingNumber(heightAttr); ok {
```

Comment:
I'd prefer `:=` instead of `=` for these two, to keep the scope the variables shorter and easier to reason about (so reader doesn't have to consider whether `w' matter before/after this chunk of code.